src/test/java/rx/internal/operators/OperatorDoOnEmptyTest.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.internal.operators;
+
+import org.junit.Test;
+import rx.Observable;
+import rx.Subscription;
+import rx.functions.Action0;
+import rx.observers.TestSubscriber;
+import rx.subjects.PublishSubject;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public final class OperatorDoOnEmptyTest {
+
+    @Test
+    public void testNonEmpty() {
+        Observable<String> source = Observable.just("Chicago", "Houston", "Phoenix");
+
+        final AtomicBoolean wasCalled = new AtomicBoolean(false);
+
+        source.doOnEmpty(new Action0() {
+            @Override
+            public void call() {
+                wasCalled.set(true);
+            }
+        }).subscribe();
+
+        assert(!wasCalled.get());
```

Comment:
Should use `assertTrue` everywhere in the unit tests rather than `assert` to ensure the assertions are run regardless of  runtime flags.